typescript/infra/src/core/govern.ts

```diff
@@ -68,12 +70,35 @@ export class AbacusCoreGovernor extends AbacusCoreChecker {
     violation: ValidatorViolation,
   ): Promise<DomainedCall> {
     const domain = violation.domain;
-    const manager = this.app.mustGetContracts(domain).validatorManager;
-    expect(manager).to.not.be.undefined;
-    const tx = await manager.populateTransaction.enrollValidator(
-      violation.data.remote,
-      violation.expected,
+    const provider = this.app.mustGetProvider(domain);
+
+    const multisigValidatorManager = MultisigValidatorManager__factory.connect(
+      violation.data.multisigValidatorManagerAddress,
+      provider,
     );
+
+    let tx: PopulatedTransaction;
+
+    // Enrolling a new validator
+    if (violation.actual === undefined && violation.expected !== undefined) {
+      tx = await multisigValidatorManager.populateTransaction.enrollValidator(
+        violation.expected,
+      );
+    } else if (
+      violation.actual !== undefined &&
+      violation.expected === undefined
+    ) {
+      // Unenrolling an existing validator
+      tx = await multisigValidatorManager.populateTransaction.unenrollValidator(
+        violation.actual,
+      );
+    } else {
+      // Invalid state
+      throw new Error(
+        `Expected exactly one of actual ${violation.actual} or expected ${violation.expected} to be undefined`,
+      );
```

Comment:
this predates me but the undefined matching looks funny
could we use an enum or sentinel?